decoder.go

```diff
@@ -0,0 +1,133 @@
+package drum
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+// DecodeFile decodes the drum machine file found at the provided path
+// and returns a pointer to a parsed pattern which is the entry point to the
+// rest of the data.
+func DecodeFile(path string) (*Pattern, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &Pattern{}
+	if err := NewDecoder(file).Decode(p); err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	return p, nil
+}
+
+// Decoder reads and decodes splice patterns from an input stream
+type Decoder struct {
```

Comment:
Yeah, but the issue with aliasing an interface is you cannot have a receive type be an interface type. So we'd have to re-write the code of how `Decoder`. We have unexported methods off `Decoder` as well as the `Decode` method that cannot be defined on a type alias on an interface.